### PR TITLE
[SPARK-35815][SQL][FOLLOWUP] Add test considering the case spark.sql.legacy.interval.enabled is true

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR adds test considering the case `spark.sql.legacy.interval.enabled` is `true` for SPARK-35815.

### Why are the changes needed?

SPARK-35815 (#33456) changes `Dataset.withWatermark` to accept ANSI interval literals as `delayThreshold` but I noticed the change didn't work with `spark.sql.legacy.interval.enabled=true`.
We can't detect this issue because there is no test which considers the legacy interval type at that time.
In SPARK-36323 (#33551), this issue was resolved but it's better to add test.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

New test.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/EventTimeWatermarkSuite.scala [hunk 1]
```diff
@@ -766,55 +766,62 @@ class EventTimeWatermarkSuite extends StreamTest with BeforeAndAfter with Matche
...
-      "interval '5' foo",
-      "interval '1 2:3:4' day to hour",
-      "interval '1 2' year to month").foreach { delayThreshold =>
-      intercept[AnalysisException] {
-        val df = MemoryStream[Int].toDF
-          .withColumn("eventTime", timestamp_seconds($"value"))
-          .withWatermark("eventTime", delayThreshold)
+  Seq("true", "false").foreach { legacyIntervalEnabled =>
+    test("SPARK-35815: Support ANSI intervals for delay threshold" +
+      s"(${SQLConf.LEGACY_INTERVAL_ENABLED.key}=$legacyIntervalEnabled") {
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/EventTimeWatermarkSuite.scala:771]:
```suggestion
      s"(${SQLConf.LEGACY_INTERVAL_ENABLED.key}=$legacyIntervalEnabled)") {
```

Comment (sarutak) [reply sql/core/src/test/scala/org/apache/spark/sql/streaming/EventTimeWatermarkSuite.scala:771]: Oops. Thanks.

**Review by HyukjinKwon — APPROVED** [review]

**Review by MaxGekk — COMMENTED** [review]
LGTM except of a minor comment.

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/EventTimeWatermarkSuite.scala [hunk 2]
```diff
@@ -766,55 +766,62 @@ class EventTimeWatermarkSuite extends StreamTest with BeforeAndAfter with Matche
...
-      "interval 4 foo",
-      "interval '5' foo",
-      "interval '1 2:3:4' day to hour",
-      "interval '1 2' year to month").foreach { delayThreshold =>
-      intercept[AnalysisException] {
-        val df = MemoryStream[Int].toDF
-          .withColumn("eventTime", timestamp_seconds($"value"))
-          .withWatermark("eventTime", delayThreshold)
+  Seq("true", "false").foreach { legacyIntervalEnabled =>
+    test("SPARK-35815: Support ANSI intervals for delay threshold" +
```
[the remark this answers is not in the document]
Comment (sarutak) [reply sql/core/src/test/scala/org/apache/spark/sql/streaming/EventTimeWatermarkSuite.scala:770]: O.K, I'll add it. Thanks.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.2. I have checked (ran the test) the PR locally.
Thank you, @sarutak and @HyukjinKwon for review.

**Pull request closed without merging**